src/cpp/shared/town/buildings.h

```diff
@@ -1,6 +1,16 @@
 #ifndef BUILDINGS_H
 #define BUILDINGS_H
 
+struct building_cost {
```

Comment:
Wow! I must have created this structure a long time ago, before I discovered the version of HEROES2W.exe with the debug symbols.

This should probably be renamed something like "SBuildingCost," in keeping with naming conventions.

Reply:
Sure I can rename it.  This name matches the type in the asm file - not sure if that mattered.

Reply:
Cool. Please do.

I spent a lot of time 2010-2012 reverse-engineering the HoMM II source code. When I discovered structures, I gave whatever names I saw fitting to them using my own naming convention.

In 2012, I discovered on GrayFace's website a version of the HoMM II binary with debug symbols. I successfully changed the HoMM II decompilation over to use them in June 2012, with the help of the bindiff tool. Upon doing so, we learned the names that the original developers used for all their classes and structures. Over the rest of the year, we converted all the names in IDA (where the names in the disassembly come from) to match the ones in the binary.

However, a few names slipped through the cracks, and this is one of them. If we were to create this structure today, the assembly name would indeed follow the naming convention of the original game.